addons/binding/org.openhab.binding.solarlog/src/main/java/org/openhab/binding/solarlog/internal/HttpUtils.java

```diff
@@ -0,0 +1,81 @@
+
+package org.openhab.binding.solarlog.internal;
+
+import java.util.concurrent.TimeUnit;
+
+import org.eclipse.jetty.client.HttpClient;
+import org.eclipse.jetty.client.api.ContentResponse;
+import org.eclipse.jetty.client.util.StringContentProvider;
+import org.eclipse.jetty.http.HttpMethod;
+import org.eclipse.jetty.http.HttpStatus;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.gson.JsonElement;
+import com.google.gson.JsonParser;
+
+/**
+ * Collection of methods to help retrieve HTTP data from a SolarLog Server
+ * Based on SqueezeBox HttpUtils, adapted for SolarLog.
+ *
+ * @author Dan Cunningham
+ * @author Svilen Valkanov - replaced Apache HttpClient with Jetty
+ * @author Johann Richard - Adapted for SolarLog Binding
+ */
+public class HttpUtils {
+    private static Logger logger = LoggerFactory.getLogger(HttpUtils.class);
+
+    private static int TIMEOUT = 5000;
+    private static HttpClient client = new HttpClient();
+    /**
+     * JSON request to get the Data from a SolarLog Device
+     */
+    private static final String JSON_REQ = "{\"801\":{\"170\":null}}";
+    private static final String URL_POSTFIX = "/getjp";
+
+    /**
+     * Simple logic to perform a post request
+     *
+     * @param url
+     * @param timeout
+     * @return
+     */
+    public static String post(String url, String postData) throws Exception {
+        if (!client.isStarted()) {
+            client.start();
+        }
+
+        // @formatter:off
+        ContentResponse response = client.newRequest(url)
+                .method(HttpMethod.POST)
+                .content(new StringContentProvider(postData))
+                .timeout(TIMEOUT, TimeUnit.MILLISECONDS)
+                .send();
+
+        int statusCode = response.getStatus();
+
+        if (statusCode != HttpStatus.OK_200) {
+            String statusLine = response.getStatus() + " " + response.getReason();
+            logger.error("Method failed: {}", statusLine);
```

Comment:
Should we log this as an error, do not throw raw type Exception but use a specific one